File: frontend/src/utils/orderStatus.ts
import type { OrderItem, OrderStatus } from '@/types'

export const orderStatusRank: Record<OrderStatus, number> = {
  IN_PROGRESS: 0,
  PENDING_COMPLETION: 0,
  PENDING_CONFIRM: 1,
  DISPUTE: 1,
  COMPLETED: 2,
  REVIEWED: 2,
  CANCELLED: 3
}

export function compareOrdersByStatus(a: Pick<OrderItem, 'status' | 'createdAt'>, b: Pick<OrderItem, 'status' | 'createdAt'>) {
  const rankDiff = orderStatusRank[a.status] - orderStatusRank[b.status]
  if (rankDiff !== 0) return rankDiff
  return new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()
}

File: backend/src/main/java/com/campushub/enums/OrderStatus.java
package com.campushub.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import lombok.Getter;

@Getter
public enum OrderStatus {
    PENDING_CONFIRM("PENDING_CONFIRM"),
    IN_PROGRESS("IN_PROGRESS"),
    PENDING_COMPLETION("PENDING_COMPLETION"),
    COMPLETED("COMPLETED"),
    CANCELLED("CANCELLED"),
    TIMEOUT("TIMEOUT"),
    DISPUTE("DISPUTE"),
    REVIEWED("REVIEWED");

    @EnumValue
    private final String value;

    OrderStatus(String value) {
        this.value = value;
    }
}
